src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 1.00f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+            case EVERETT_TRANSIT:
+            case SEATTLE_STREET_CAR:
+                return 1.50f;
+            case PIERCE_COUNTY_TRANSIT:
+            default:
+                return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca senior discount fares based on the fare and ride types.
+     */
+    private float getSeniorFare(Fare.FareType fareType, RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case EVERETT_TRANSIT:
+                return fareType.equals(Fare.FareType.orcaSenior) ? 0.50f : defaultFare;
+            case PIERCE_COUNTY_TRANSIT:
+            case SEATTLE_STREET_CAR:
+            case KITSAP_TRANSIT:
+                // Pierce, Seattle Streetcar, and Kitsap only provide discounted senior fare for orca.
+                return fareType.equals(Fare.FareType.orcaSenior) ? 1.00f : defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 3.00f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 2.50f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+                return 1.00f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca youth discount fares based on the ride type.
+     */
+    private float getYouthFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.75f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 3.00f;
+            case PIERCE_COUNTY_TRANSIT: return 1.00f;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 2.00f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+            case EVERETT_TRANSIT:
+            case SEATTLE_STREET_CAR:
+                return 1.50f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        List<Ride> ridesSingleton = new ArrayList<>();
+        ridesSingleton.add(ride);
```

Comment:
```suggestion
        List<Ride> ridesSingleton = Lists.newArrayList(ride);
```
and you could even inline `ridesSingleton`.